### DISCUSSION TITLE
Viewers Not Connecting Via Multicast

### DISCUSSION THREAD

**Discussion by gfaulk09** [bluenviron/mediamtx#2448] [General]
Trying to use MediaMTX with video storm IP HDMI matrix software. We have over 30tvs. Other sources/software multicast throughout our network properly without issue.
It seems that connections to the tvs using MediaMTX are connecting just not with multicast which is the issue.
We are using OBS to publish the stream to MediaMTX. Android TV's (mix of Nvidia Shields, Chromecasts) are running the NetPlayTV application and are playing the streams on the connected tvs.
Attached below are log files 
[log.txt](https://github.com/bluenviron/mediamtx/files/12785008/log.txt)

**Comment by aler9** [comment]
Hello, your readers are not using the UDP-multicast transport protocol, but the plain UDP transport protocol. This line:

```
2023/10/02 12:31:58 INF [RTSP] [session 607ba9a7] is reading from path 'mystream', with UDP, 2 tracks (H264, MPEG-4 Audio)
```

Should be

```
2023/10/02 12:31:58 INF [RTSP] [session 607ba9a7] is reading from path 'mystream', with UDP-multicast, 2 tracks (H264, MPEG-4 Audio)
```

 in RTSP, the transport protocol is chosen by clients, not by the server.


In order to change the transport protocol, you have to tune your client, that in this case is NetPlayTV. It has to support UDP-multicast and has to provide an option to change the protocol. Since it's apparently based on FFmpeg's libavformat, it should support it.

**Reply by aler9** [reply]
Another way consists in disabling all transport protocols except multicast in the server configuration:

```yml
protocols: [multicast]
```

[a comment by an automated account is not shown]
